libmscore/figuredbass.cpp

```diff
@@ -1080,7 +1080,7 @@ void FiguredBass::read(XmlReader& e)
       while (e.readNextStartElement()) {
             const QStringRef& tag(e.name());
             if (tag == "ticks")
-                  setTicks(Fraction::fromTicks(e.readInt()));
+                  setTicks(e.readFraction());
```

Comment:
This should probably break reading scores created in versions 3.0.5 and below though. As far as I remember, we have intentionally not switched to fractions in `ticks` or some similar tag in order to retain a compatibility with the older versions of MuseScore. So maybe we should write integer ticks value instead or, if new scores happen to be already incompatible with 3.0.5 version, at least read this tag's value in a way that would automatically detect whether it contains a fraction or an integer value.

Reply:
Scores saved more than once with 3.1 or later are already broken (save 1: fraction is stored in file; read 1: an integer, usually 0, is read back; save 2: "0/&lt;something&gt;" is written in the file and the datum is gone).

Scores created before 3.1 are broken after a similar path under 3.1 or later, which is not unlikely to happen during the time from 3.1 release to next release. So, it is my impression than things are already broken in the majority of cases (chances are, if one needs F.B., he uses it often).

I think that things should be improved at a lower level: `XmlReader::readFraction()` should read both explicit fractions as well as integers, as integers **are** fractions!

1) Shall I add this change too to this PR?
2) Shall I do it in a separate PR?
3) Shall I leave the matter to someone else?

Reply:
Yes, integers are fractions but in this context integers and fractions have different meaning, so it would be good to handle them separately depending on their type. Things may have been already broken for scores that are being actively worked on, but there may be scores that were finished before 3.1 release and stored on musescore.com or elsewhere. It would be better to maintain a compatibility with their file format if possible, at least during 3.x series life cycle.

That is why I propose adding something separate, like `XmlReader::readTicks()`, which would handle ticks values properly depending on what has actually been written to a file. As this concerns file format compatibility, I believe it would be better to do this as a part of this PR.

Reply:
Well, I looked into the thing and `XmlReader::readFraction()` already knows about ticks: it already retrieves a fraction and, if there is no denominator, it converts a tick value to a fraction; see [code here](https://github.com/musescore/MuseScore/blob/master/libmscore/xmlreader.cpp#L177).

So, this PR should already work with all ver. 3 scores, either from before or after the change from ticks to fractions.

If the behaviour of `XmlReader::readFraction()` needs to be improved (which is always possible), I think this would belong to a separate, specific, PR (and issue in the issue tracker).

Reply:
Well, then this PR should indeed work for old scores as well, thanks for looking at this! I just did not expect that the function named `readFraction()` would also read integer ticks, but any changes to that would indeed belong to a separate issue.